CANTroller2/src/ErrorManager.h

```diff
@@ -0,0 +1,78 @@
+#ifndef ERROR_MANAGER_H
+#define ERROR_MANAGER_H
+
+#include <Arduino.h>
+#include <string>
+#include <SD.h>
+
+enum class ErrorType {
+    SENSOR_DISCONNECTED,
+    TEMPERATURE_WARNING,
+    TEMPERATURE_ALARM,
+    // Add more error types as needed
+};
+
+class ErrorManager {
+public:
+    void log_error(ErrorType error, const std::string& source, const std::string& message) {
+        // Log the error with the source, message, and error type
+        unsigned long timeSinceBoot = millis();
+        Serial.printf("Time: %lu, Error: %s, Source: %s, Message: %s\n", timeSinceBoot, error, source.c_str(), message.c_str());
+        log_error_to_sd_card(error, source, message, timeSinceBoot);
+    }
+
+   void log_error_to_sd_card(ErrorType error, const std::string& source, const std::string& message, unsigned long timeSinceBoot) {
+        static bool firstLog = true;
+        static String logFileName;
+
+        // create a new log file for each time we boot up
+        if (firstLog) {
+            unsigned long bootTime = millis();
+            logFileName = "log_" + String(bootTime) + ".txt";
+            firstLog = false;
+        }
+
+        File logFile = SD.open(logFileName.c_str(), FILE_WRITE);
```

Comment:
will this append? do we ever clear the log? and do we have a way to read the log?

Reply:
It does append yes, and I'm assuming we'd open the file on a laptop by popping the SD card out. I could make other options, like we could make a function that prints out the most recent log file to the serial monitor on boot